### PR TITLE
Fix default workspace class is not selected

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Show the workspace class that will be used if no workspace class has been selected yet by the user. 

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/12435

## How to test
- Open the workspace settings in the preview environment
- Check that Default is selected even though now workspace class has been selected yet

## Release Notes

```release-note
None
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@Furisto: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/service/service-mock.ts [hunk 1]
```diff
@@ -265,6 +266,9 @@ const gitpodServiceMock = createServiceMock({
             },
         };
     },
+    getBillingModeForUser: async () => {
```
Comment (easyCZ) [inline components/dashboard/src/service/service-mock.ts:269]: This seems unrelated to the PR, why is this change needed?

Comment (Furisto) [reply components/dashboard/src/service/service-mock.ts:269]: The settings page calls this method and if it is not there the page cannot be displayed when using the service mock.

**Review by easyCZ — APPROVED** [review]
LGTM
Will require a rebase
